File: app/src/main/java/com/coderstory/flyme/xposed/IModule.java
package com.coderstory.flyme.xposed;

import java.lang.reflect.InvocationTargetException;

import de.robv.android.xposed.IXposedHookZygoteInit;
import de.robv.android.xposed.callbacks.XC_InitPackageResources;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

public interface IModule {
    void handleInitPackageResources(XC_InitPackageResources.InitPackageResourcesParam resparam);

    void handleLoadPackage(XC_LoadPackage.LoadPackageParam lpparam) throws IllegalAccessException, InvocationTargetException, InstantiationException;

    void initZygote(IXposedHookZygoteInit.StartupParam startupParam);
}

File: app/src/main/java/com/coderstory/flyme/xposed/IModule.kt
package com.coderstory.flyme.xposed

import de.robv.android.xposed.IXposedHookZygoteInit.StartupParam
import de.robv.android.xposed.callbacks.XC_InitPackageResources.InitPackageResourcesParam
import de.robv.android.xposed.callbacks.XC_LoadPackage.LoadPackageParam
import java.lang.reflect.InvocationTargetException

interface IModule {
    fun handleInitPackageResources(resparam: InitPackageResourcesParam)

    @Throws(IllegalAccessException::class, InvocationTargetException::class, InstantiationException::class)
    fun handleLoadPackage(lpparam: LoadPackageParam)
    fun initZygote(startupParam: StartupParam?)
}